### ISSUE TITLE
Request for clarification

### ISSUE THREAD

**Issue by c0mmand3r420** [grcninja/N0D#1]
UP/DOWN - Is this referring to the HTTP return header on port 80? What if the web server is running on a non standard port? 8081? 88? 81? etc etc.
    Having an IP address should be enough of an 'up' verification.

What if the domain is not currently associated with an IP address? Is it ignored, only outputting 'live' domains when the script is ran?

   A step is missing here. Before creating a removal rule, we need to know what the IP for that domain was originally.

Also, what happens when a single domain cycles through multiple IP Addresses? 
  If it is to be unblocked @ null0.  1.4 (listed as later development stage) would be helpful at this stage.
      Example: abc.com = ip 1.1.1.1    - blocked.  second scan 1 week later. abc.com = 1.2.2.1
        Are both now blocked?
       Or is 1.1.1.1 unblocked before adding 1.2.2.1?
        When abc.com is removed from the blocklist, which ip is unblocked? The last ? Both? How are those determined by just the domain name from the blocklist file? 
          How do you know what that IP address is?
          Are all previously written status files being read here? Do we use IPs from the last created, or all?

Lots of questions I know. Just throwing them all at you so they can be addressed in detail within a project guideline / project plan file for further functionality clarification.

**Comment by grcninja** [comment]
So there's some basic code out to @Lojikil for review right now.  But
here's some answers:

UP/DOWN - Is this referring to the HTTP return header on port 80? What is
the web server is running on a non standard port? 8081? 88? 81? etc etc.
Having an IP address should be enough of an 'up' verification.
ANSWER:  the current code set is using the socket.gethostbyaddr() so we
aren't going to specify port as it's already part of the function

---

---

---

What if the domain is not currently associated with an IP address? Is it
ignored, only outputting 'live' domains when the script is ran?ANSWER: If
it does not respond, it is appended to a list of non-responsive domains

---

---

---

A step is missing here. Before creating a removal rule, we need to know
what the IP for that domain was originally.
ANSWER: I have no idea what you mean with this sentence

---

---

---

Also, what happens when a single domain cycles through multiple IP
Addresses?
If it is to be unblocked @ null0. 1.4 (below) would be helpful at this
stage.
Example: abc.com = ip 1.1.1.1 - blocked. second scan 1 week later. abc.com =
1.2.2.1
Are both now blocked?
Or is 1.1.1.1 unblocked before adding 1.2.2.1?
ANSWER: YES because many of these sites leverage a dynamic dns service.

---

---

---

When abc.com is removed from the blocklist, which ip is unblocked? The last
? Both? How are those determined by just the domain name from the blocklist
file?
How do you know what that IP address is?
Are all previously written status files being read here? Do we use IPs from
the last created, or all?

ANSWER:  When a domain is removed from the source, all the IPs for that
domain will be unblocked.  Then the module to check the new list will run,
if the IP has been reassigned to a bad domain then it will be added back.
Basically, the router would be updated first with the file to remove null
routes that aren't needed then add null routes that are needed.  So the
DENY's are always loaded last.

---

---

---

The point is that it is NEVER EVER going to be a perfect solution because
IPs can change 10 minutes after we check for them, but it will always be
the best possible solution.

---

---

---

Lots of questions I know. Just throwing them all at you so they can be
addressed in detail within a project guideline / project plan file for
further functionality clarification.
